### PR TITLE
chore: improve tests coverage

### LINKED ISSUE #113: [Task] Identify and add missing test cases
## Summary

The current test suite does not fully cover all modules, code paths, and error conditions in the library.
This issue aims to identify missing test cases and add them to improve overall reliability and confidence during refactoring.

## Scope

This task includes:

- Reviewing the existing test suite to determine what is missing
- Adding test cases for uncovered normal‑flow behavior across the library
- Adding test cases for edge cases and unusual inputs

## Motivation

- Improve confidence in ongoing refactoring for v2.0.0
- Ensure that both normal and abnormal inputs are handled correctly
- Detect regressions earlier

## Discussion points

- Which modules or functions currently lack coverage
- Which error conditions should be explicitly tested

## Notes on abnormal‑case testing

This issue explicitly includes adding tests for:

- Invalid or malformed inputs
- Boundary conditions
- Cases where exceptions should be raised
- Unexpected or inconsistent internal states
- Inputs that violate assumptions but should still fail gracefully

### PR DESCRIPTION
**Pull request by Nihisil**
Closes #113

Added 100% tests coverage.

I have validated that yaku calculation, agari and shanten are working as before for 2kk hands examples.

### CODE REVIEW

File: mahjong/hand_calculating/hand.py [hunk 1]
```diff
@@ -493,27 +483,15 @@ def estimate_hand_value(
...
                     config.yaku.dora.han_open = count_of_dora
                     config.yaku.dora.han_closed = count_of_dora
                     hand_yaku.append(config.yaku.dora)
                     han += count_of_dora
 
-                if count_of_aka_dora:
-                    config.yaku.aka_dora.han_open = count_of_aka_dora
-                    config.yaku.aka_dora.han_closed = count_of_aka_dora
-                    hand_yaku.append(config.yaku.aka_dora)
-                    han += count_of_aka_dora
```
Comment (Nihisil) [inline mahjong/hand_calculating/hand.py]: kokushi hand can't have aka dora

File: tests/tests_utils.py [hunk 2]
```diff
@@ -251,3 +256,98 @@ def test_is_dora_indicator_for_terminal(sou: str, pin: str, man: str, honors: st
...
+def test_meld_str() -> None:
+    tiles = TilesConverter.string_to_136_array(man="123")
+    meld = Meld(meld_type=Meld.CHI, tiles=tiles)
+    assert str(meld) == f"Type: chi, Tiles: 123m {tiles}"
+
+
+def test_meld_repr() -> None:
+    tiles = TilesConverter.string_to_136_array(man="111")
+    meld = Meld(meld_type=Meld.PON, tiles=tiles)
+    assert repr(meld) == str(meld)
```
Comment (Apricot-S) [inline tests/tests_utils.py]: Move them to `tests_meld.py`.
